.github/workflows/pre-commit.yaml

```diff
@@ -10,5 +10,14 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
-      - uses: pre-commit/action@v2.0.3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Run image
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: '1.3.2'
+      - name: Install dependencies
+        run: poetry install
+      - name: Run pre-commit
+        run: poetry run pre-commit run --all-files
```

Comment:
Why running via poetry instead of via `pre-commit/action@v2.0.3 `

Reply:
The reason behind it is that now we need to have all the packages installed to have their types for the typecheck. And it seems like `pre-commit/action@v2.0.3` is not configurable to work with poetry environment.